src/sauce-testreporter.js

```diff
@@ -137,6 +170,41 @@ exports.sauceReporter = async (browserName, assets, results) => {
     console.warn("Failed to prepare test", e);
   }
 
+  return sessionId;
+}
+
+exports.sauceReporter = async (browserName, assets, results) => {
+// SAUCE_JOB_NAME is only available for saucectl >= 0.16, hence the fallback
+  const testName = process.env.SAUCE_JOB_NAME || `DevX TestCafe Test Run - ${(new Date()).getTime()}`;
+
+  let tags = process.env.SAUCE_TAGS
+  if (tags) {
+    tags = tags.split(",")
+  }
+
+  let build = process.env.SAUCE_BUILD_NAME
+
+  /**
+   * replace placeholders (e.g. $BUILD_ID) with environment values
+   */
+  const buildMatches = (build || '').match(/\$[a-zA-Z0-9_-]+/g) || []
+  for (const match of buildMatches) {
+    const replacement = process.env[match.slice(1)]
+    build = build.replace(match, replacement || '')
+  }
+
+  let sessionId;
+  if (process.env.ENABLE_PLATFORM === 'true') {
+    sessionId = await createJobShell(api, testName, browserName, tags)
+  } else {
+    sessionId = await createJobLegacy(api, region, browserName, testName, tags, build)
+  }
+
+  if (undefined === sessionId || 0 === sessionId) {
```

Comment:
It could be simplified
```suggestion
  if (!sessionId) {
```